app/mixins/navigation.js

```diff
@@ -341,5 +341,20 @@ export default Ember.Mixin.create({
 
       return nav;
     });
-  })
+  }),
+
+  findNavItemByRoute: function(route) {
+    return this.navItems.find(function(el) {
```

Comment:
I don't think you want to use find here.  According to the docs 
[find](http://emberjs.com/api/classes/Ember.Array.html#method_find) "Returns the first item in the array for which the callback returns true".  You are not returning true anywhere here and the way this code is written, if the item is found in a subnav item, it is going to return the parent item, not the subnav item.  You could accomplish what you need with forEach loops.